pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -220,11 +224,208 @@ def _get_fourier_models_data(self) -> pd.DataFrame:
             n_order=self.yearly_seasonality,
         )
 
+    def channel_contributions_forward_pass(
+        self, channel_data: Union[pd.DataFrame, pd.Series, npt.NDArray[np.float_]]
+    ) -> npt.NDArray[np.float_]:
+        """Evaluate the channel contribution for a given channel data and a fitted model, ie. the forward pass.
+
+        Parameters
+        ----------
+        channel_data : array-like
+            Input channel data.
+
+        Returns
+        -------
+        array-like
+            Transformed channel data.
+        """
+        alpha_posterior = self.fit_result["posterior"]["alpha"].to_numpy()
+
+        lam_posterior = self.fit_result["posterior"]["lam"].to_numpy()
+        lam_posterior_expanded = np.expand_dims(a=lam_posterior, axis=2)
+
+        beta_channel_posterior = self.fit_result["posterior"]["beta_channel"].to_numpy()
+        beta_channel_posterior_expanded = np.expand_dims(
+            a=beta_channel_posterior, axis=2
+        )
+
+        geometric_adstock_posterior = geometric_adstock(
+            x=channel_data,
+            alpha=alpha_posterior,
+            l_max=self.adstock_max_lag,
+            normalize=True,
+            axis=0,
+        )
+
+        logistic_saturation_posterior = logistic_saturation(
+            x=geometric_adstock_posterior,
+            lam=lam_posterior_expanded,
+        )
+
+        channel_contribution_forward_pass = (
+            beta_channel_posterior_expanded * logistic_saturation_posterior
+        )
+        return channel_contribution_forward_pass.eval()
```

Comment:
Not specific to this PR, but some things to keep in mind:
1. If you are doing a single call, and compilation takes a considerably chunk, you can consider setting the mode to `FAST_COMPILE` and/or disable C compilation via `config.cxx=""`. 
2. If a user is likely to call this method many times in a single workflow, it may make sense to compile a generic function and cache it. In subsequent calls one could just pass the new inputs to the function without having to recompile it every time.